sum.go

```diff
@@ -103,17 +70,24 @@ func isBlake2b(code uint64) bool {
 	return code >= BLAKE2B_MIN && code <= BLAKE2B_MAX
 }
 
-func sumBlake2b(size uint8, data []byte) []byte {
-	hasher, err := blake2b.New(&blake2b.Config{Size: size})
+func sumBlake2s(data []byte, size int) ([]byte, error) {
+	if size != 32 {
+		return nil, fmt.Errorf("unsupported length for blake2s: %d", size)
+	}
+	d := blake2s.Sum256(data)
+	return d[:], nil
+}
+func sumBlake2b(data []byte, size int) ([]byte, error) {
+	hasher, err := blake2b.New(&blake2b.Config{Size: uint8(size)})
 	if err != nil {
-		panic(err)
+		return []byte{}, err
```

Comment:
This should probably return `nil`, not `[]byte{}`.

Reply:
Agree, thank you!